lib/Doctrine/deploy/AddServiceGroupRoles.php

```diff
@@ -0,0 +1,93 @@
+<?php
+
+require_once __DIR__ . "/../bootstrap.php";
+require_once __DIR__ . "/AddUtils.php";
+
+$usersRolesFileName = __DIR__ . "/" . $GLOBALS['dataDir'] .
+    "/UsersAndRoles.xml";
+$usersRoles = simplexml_load_file($usersRolesFileName);
+
+foreach ($usersRoles as $user) {
+    foreach ($user->USER_ROLE as $role) {
+        // Check for blank role, skip if it's blank
+        if ((string) $role->USER_ROLE == "") {
+            continue;
+        }
+
+        // Skip all non-serviceGroup roles
+        if ((string) $role->ENTITY_TYPE !== "serviceGroup") {
+            continue;
+        }
+
+        // get roletype entity
+        $userRole = (string) $role->USER_ROLE;
+        $dql = "SELECT rt FROM RoleType rt WHERE rt.name = :roleType";
+        $roleTypes = $entityManager->createQuery($dql)
+                                   ->setParameter(':roleType', $userRole)
+                                   ->getResult();
+
+        /*
+         * Error checking: ensure each role type refers to exactly
+         * one role type
+         */
+        if (count($roleTypes) !== 1) {
+            throw new Exception(count($roleTypes) . " role types " .
+                "found with name: " . $userRole);
+        }
+
+        foreach ($roleTypes as $result) {
+            $roleType = $result;
+        }
```

Comment:
What is this (and similar) for loop doing?

Reply:
These loops are present in similar files such as https://github.com/GOCDB/gocdb/blob/dev/lib/Doctrine/deploy/AddNGIRoles.php from which I copied them over. Deploying the sample data doesn't work without them, and I think they are there as the corresponding database query above returns an array, but only with one item in it. The foreach loop takes the one data item in the array and makes it a normal string, which is then used later on to make the role in the Role() function.

Reply:
ah okay - I think something like `$roleType = $roleTypes[0]` would be clearer as to what is happening. Could you change that for AddServiceGroupRoles.php?

Reply:
ok

Reply:
done